src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -1607,5 +1607,88 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other,
                 value, comparer);
             return BinarySearch(span, comparable);
         }
+
+
+        public static void Sort<T>(this Span<T> array)
+        {
+            if (array == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            Sort<T>(array!, 0, array!.Length, null); // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+        }
+
+        public static void Sort<TKey, TValue>(this Span<TKey> keys, Span<TValue> items)
+        {
+            if (keys == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.keys);
+            Sort<TKey, TValue>(keys!, items, 0, keys!.Length, null); // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+        }
+
+        public static void Sort<T>(this Span<T> array, int index, int length)
+        {
+            Sort<T>(array, index, length, null);
+        }
+
+        public static void Sort<TKey, TValue>(this Span<TKey> keys, Span<TValue> items, int index, int length)
+        {
+            Sort<TKey, TValue>(keys, items, index, length, null);
+        }
+
+        public static void Sort<T>(this Span<T> array, System.Collections.Generic.IComparer<T>? comparer)
+        {
+            Sort<T>(array, 0, array.Length, comparer); 
+        }
+
+        public static void Sort<TKey, TValue>(this Span<TKey> keys, Span<TValue> items, System.Collections.Generic.IComparer<TKey>? comparer)
+        {
+            Sort<TKey, TValue>(keys, items, 0, keys.Length, comparer);
+        }
+
+        public static void Sort<T>(this Span<T> array, int index, int length, System.Collections.Generic.IComparer<T>? comparer)
```

Comment:
I do not see overloads that take Span, index and length in the approved API list.

Reply:
Sorry, I wasn't clear, Span API has not been implemented yet and I haven't clean the code up yet either. I wanted to make sure Array code was viable first, there are other changes needed to support the full Span API as well.